.github/workflows/build-flutter.yml

```diff
@@ -1,121 +1,175 @@
+              version_string = `${version}`
+            }
+            const newPubspec = pubspec.replace(/version: .*/, `version: ${version_string}`)
+            fs.writeFileSync(pubspecPath, newPubspec)
+
+      - name: Install Dependencies (linux)
+        shell: sh
+        if: contains(inputs.targets, 'appimage')
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install -y make cmake ninja-build libgtk-3-dev locate libfuse2 libsecret-1-dev libjsoncpp-dev
+          wget -O appimagetool "https://github.com/AppImage/AppImageKit/releases/download/continuous/appimagetool-x86_64.AppImage"
+          chmod +x appimagetool
+          mv appimagetool /usr/local/bin/
+    
+      - name: Install Dependencies (macos)
+        shell: sh
+        if: contains(inputs.targets, 'dmg')
+        run: |
+          npm install -g appdmg
+
+      - name: Build App for ${{ inputs.platform }}
+        run: |
+          echo "Building for platform: ${{ inputs.platform }}"
+          echo "Version: ${{ inputs.version }}"
+          echo "Build ID: ${{ inputs.build }}"
+          echo "Targets: ${{ inputs.targets }}"
+
+          dart pub global activate flutter_distributor
+
+          platform=${{ inputs.platform }}
+          IFS=',' read -r -a flutter_extra_args <<< "${{ inputs.flutter-extra-args }}"
+          IFS=',' read -r -a targets <<< "${{ inputs.targets }}"
+
+          if [ "${#targets[@]}" -gt 0 ]; then
+            echo "No targets specified. Exiting..."
+            exit 1
+          fi
+
+          # If flutter_extra_args has split-per-abi,
+          # separate APK build than other targets 
+          # and use split-per-abi only on android apk build.
+          if [ "${platform}" = "android" ]; then
+            if [[ " ${flutter_extra_args[@]} " =~ " split-per-abi " ]]; then
+              echo "WARNING: Splitting per ABI only work on APK targets. Removing it from other android targets."
+              flutter_distribution package --flutter-extra-args=$(IFS="," echo "${flutter_extra_args[*]}") --platform android --targets apk
+              build_path="build/app/outputs/flutter-apk"
+              output_path="dist/${{ inputs.version }}+${{ inputs.build }}"
+              wrong_file_path=$(ls ${output_path}/*.apk | head -n1)
+              generated_name=$(basename "${wrong_file_path%-*}")
+              rm ${output_path}/*-${{ inputs.version }}+${{ inputs.build }}-*.apk
+              # for each in build/app/outputs/flutter-apk/*.apk
+              flutter_extra_args=(${flutter_extra_args[@]//*split-per-abi*/})
+              targets=(${targets[@]//*apk*/})
+            fi
+          fi
 
-      - name: Upload Artifacts
+          # if targets is not empty, build the targets
+          if [ "${#targets[@]}" -gt 0 ]; then
+            flutter_distributor package --flutter-extra-args=$(IFS="," echo "${flutter_extra_args[*]}") --platform ${{ inputs.platform }} --targets $(IFS="," echo "${targets[*]}")
```

Comment:
Does it publish to the stores ?

Reply:
No, flutter distributor allow to build multiple targets at once.